DAUB-29

jobguitiche opened the pull request:
## Description
This PR check if `DataElement` have `FileResources` data and if exist use them to show `MediaDialogFragment`. If not, show a simple `CostumDialog`.

[ jira issue ](https://dhis2-mobile-ui.atlassian.net/browse/DAUB-29?atlOrigin=eyJpIjoiOTBmODlhNTJiYTVmNDk1Y2IwNWY1YjY4ZGYwZmFkZmMiLCJwIjoiaiJ9)

## Solution description
- The key to understand this solution is this code:
`val dataElement = getD2().dataElement(intent.uid)`
Using this code, I was able to access the `DataElement` data after hours of debugging.

## Covered unit test cases
- None

## Where did you test this issue?
- [ ] Smartphone Emulator
- [x] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [x] 11.X - 13.X
- [ ] Other
## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

jobguitiche committed:
feat: update 'showDescriptionLabelDialog()' function

jobguitiche committed:
feat: uses data element data

jobguitiche committed:
feat: code optimizations

jobguitiche committed:
refactor: ktlint format

jobguitiche committed:
Merge branch 'develop' into DAUB-29

NelsonChad commented on form/src/main/java/org/dhis2/form/ui/FormView.kt:
```diff
@@ -733,32 +732,35 @@ class FormView : Fragment() {
-                    )
-                }
-            }
+        if (dataElement.valueType()?.isFile == true) {
```
Must check if the dataElement uid exists in dataStore, if yes, show the custom dialog else show the default dialog

NelsonChad commented on form/src/main/java/org/dhis2/form/ui/FormView.kt:
```diff
@@ -733,32 +732,35 @@ class FormView : Fragment() {
-            .show(childFragmentManager, "ComposeDialog")
-    }
+    private fun showDialog(intent: RecyclerViewUiEvents.ShowDescriptionLabelDialog) {
+        val dataElement = getD2().dataElement(intent.uid)
```
intent.uid is enough to compare, it holds the id, don't need go get the entire dataElement.

jobguitiche committed:
Merge branch 'develop' into DAUB-29

jobguitiche committed:
feat: testing data store implementation

jobguitiche committed:
refactor: simple improvements

jobguitiche committed:
Squashed commit of the following:

commit 7465ac9993826a64a8801175d0bbfe63931d6b90
Author: Nelson  Chadali <nchadaly@gmail.com>
Date:   Tue Oct 31 03:38:50 2023 +0200

    Daub 25-  download media (#18)

    * Configurando os dados para baixar media

    * Add download media feature

    * fix download method and getLocalFile

    * remove comments and uneccessary code

    * add return to getLocalMedia and remove aneccessary code]

    ---------

    Co-authored-by: Miguel Vasco Macamo <miguelvascomacamo@gmail.com>

jobguitiche committed:
feat: working code!

jobguitiche committed:
Merge branch 'develop' into DAUB-29

jobguitiche committed:
feat: more uids tests

jobguitiche committed:
refactor: basic clean up

jobguitiche committed:
feat: implements loading media dialog

jobguitiche committed:
feat: improves media dialog

jobguitiche committed:
DAUB-29 improvements (#22)

* fix 404 request error exception

* wip: improve loading state

* fix: error showing media dialog

- 'checkDataElement()' improvement
- Changes 'println(x)' to 'timber.d(x)
- General code clean up
- fix bugs on calling 'MediaDialog'

* feat: show error message for invalid media file paths

* wip: split functions to reformat the code

* wip: clean code

* fix: multiples dialogs appears after second click

* refactor: change function access

- 'getFileExtension()'
- 'downloadMedia()'

* refactor: clean up 'downloadMedia()' function

* feat: create a function 'createDownloadDirectory()'

* refactor: create 'createMediaFilePath()' function

* refactor: adds 'saveMediaToFile()' function

---------

Co-authored-by: Nelson  Chadali <nchadaly@gmail.com>
